src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateParts = {};
+  const dateValues = date.split(fromSeparator);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateParts[fromFormat[i]] = dateValues[i];
+  }
+
+  if (dateParts['YY']) {
+    const year = parseInt(dateParts['YY'], 10);
+
+    if (year < 30) {
+      dateParts['YYYY'] = `20${dateParts['YY']}`;
+    } else {
+      dateParts['YYYY'] = `19${dateParts['YY']}`;
+    }
+  }
```

Comment:
The condition to check for 'YY' and convert it to 'YYYY' is not handling the case where the 'toFormat' does not require a 'YYYY' part. You should only perform this conversion if 'YYYY' is actually part of the 'toFormat' array.